File: app/src/main/java/com/junli/android_database/MainActivity.java
package com.junli.android_database;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.junli.android_database.java.room.DatabaseRoomAssistant;
import com.junli.android_database.java.room.domain.NormalUser;
import com.junli.android_database.kotlin.room.DatabaseRoomCreator;

public class MainActivity extends AppCompatActivity {
    @Override
    public boolean navigateUpTo(Intent upIntent) {
        return super.navigateUpTo(upIntent);
    }

    @SuppressLint("MissingInflatedId")
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button createKotlinDBBtn = findViewById(R.id.create_kotlin_database_btn);

        createKotlinDBBtn.setOnClickListener(v -> {
            new Thread(() -> {
                DatabaseRoomCreator databaseRoomCreator = DatabaseRoomCreator.Companion.getDataBaseInstance(getApplicationContext());
                databaseRoomCreator.getNormalUserApi().insertNormalUser(
                        new com.junli.android_database.kotlin.room.domain.NormalUser(null, "LiJun", "JunLi")
                );
            }).start();
            new Thread(() -> {
                DatabaseRoomAssistant databaseRoomAssistant = DatabaseRoomAssistant.getDataBaseInstance(getApplicationContext());
                databaseRoomAssistant.getNormalUserDao().insertNormalUser(new NormalUser("LiJun2", "JunLi2"));
            }).start();
        });
    }
}

File: kotlin-room/kotlin-room/src/main/java/com/junli/kotlin_room/MainActivity.kt
package com.junli.kotlin_room

import android.annotation.SuppressLint
import android.content.ContentValues.TAG
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.junli.kotlin_room.room.DatabaseRoomCreator
import com.junli.kotlin_room.room.domain.NormalUser

class MainActivity : AppCompatActivity() {
    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btn = findViewById<Button>(R.id.button)
        val tv = findViewById<TextView>(R.id.tv_sqlite_return)
        val databaseRoomCreator = DatabaseRoomCreator.getDataBaseInstance(applicationContext)
        val normalUserDao = databaseRoomCreator.getNormalUserApi()

        btn.setOnClickListener {
            Thread {
                var i = 0
                repeat(5) {
                    normalUserDao.insertNormalUser(
                        NormalUser(
                            null,
                            "LiJun Kotlin $i",
                            "JunLi Kotlin ${i++}"
                        )
                    )
                }
            }.start()
        }
        tv.setOnClickListener {
            Log.d(TAG, "点击了")
            Thread {
                val normalUserList = normalUserDao.selectAllNormalUser()
                val stringBuilder: StringBuilder = StringBuilder()
                normalUserList.forEach {
                    stringBuilder.append("UserID:${it.userID} UserName:${it.userName} NickName:${it.nickName}")
                        .append("\n")
                }
                runOnUiThread {
                    tv.text = stringBuilder
                }
            }.start()
        }

    }
}